docs/developer/community/contributing.mdx

```diff
@@ -88,6 +242,165 @@ Most editors and IDEs support this file either directly or via plugins. See the
+First, create a new enum in the `error_codes.py` file in the main app directory.
+Then use it to create graphql enum in `saleor/graphql/core/enums.py` file.
+Now you are ready to define the error class. The class must inherit from `Error` and
+must have a `code` field. Any other are optional but could be useful to specify
+what inputs raised an error. Here is an example:
+```python
+class AppError(Error):
+    code = AppErrorCode(description="The error code.", required=True)
+    permissions = NonNullList(
+        PermissionEnum,
+        description="List of permissions which causes the error.",
+        required=False,
+    )
+```
+
+
+
+
+#### Sorting and filtering
+To allow sorting the queryset you must create the `SortInputObjectType` and 
+corresponding sorting enum, and it should go to the dedicated `sorters.py` file in
+the given app module in `API`.
+
+```python
+class AppSortField(graphene.Enum):
+    NAME = ["name", "pk"]
+    CREATION_DATE = ["created", "name", "pk"]
+
+    @property
+    def description(self):
+        if self.name in AppSortField.__enum__._member_names_:
+            sort_name = self.name.lower().replace("_", " ")
+            return f"Sort apps by {sort_name}."
+        raise ValueError("Unsupported enum value: %s" % self.value)
+
+
+class AppSortingInput(SortInputObjectType):
+    class Meta:
+        sort_enum = AppSortField
+        type_name = "apps"
+```
+:::warning
+Remember, the list with sorting order must have a unique field.
+:::
+
+
+Similar situation is for filtering, you need to create `FilterInputObjectType`
+and django `FilterSet` in dedicated `filters.py` file.
+```python
+class AppFilterInput(FilterInputObjectType):
+    class Meta:
+        filterset_class = AppFilter
+
+
+class AppFilter(django_filters.FilterSet):
+    type = EnumFilter(input_class=AppTypeEnum, method=filter_app_type)
+    search = django_filters.CharFilter(method=filter_app_search)
+    is_active = django_filters.BooleanFilter()
+
+    class Meta:
+        model = models.App
+        fields = ["search", "is_active"]
+```
+
+Next, you need to create `CountableConnection` for given query:
+```python
+class AppCountableConnection(CountableConnection):
+    class Meta:
+        node = App
+```
+
+Then in the `schema.py` file, you need to change the corresponding query.
+It must be a `FilterConnectionField` with created connection field, 
+sorting input must be used in `sort_by` argument, and filter input in the `filter`
+argument of the query:
+
+```python
+class AppQueries(graphene.ObjectType):
+    apps = FilterConnectionField(
+        AppCountableConnection,
+        filter=AppFilterInput(description="Filtering options for apps."),
+        sort_by=AppSortingInput(description="Sort apps."),
+        description="List of the apps.",
+    )
+```
+
+#### How to handle breaking changes?
+- provide compatibility layer by supporting two solutions for a specified time period and removing it in next major version
```

Comment:
If it has been decided to use commas above(e.g. 39-44 lines) then please also add here.